### DISCUSSION TITLE
RPT With Degradation Experiment

### DISCUSSION THREAD

**Discussion by MirAbbasAli2A** [pybamm-team/PyBaMM#3481] [Q&A] [answered]
Hello - 

I am having some difficulty plotting an RPT with degradation experiment. I was following the example notebook on RPT with degradation and wrote the following code:

import pybamm
import matplotlib.pyplot as plt
import numpy as np
from os import path
import pandas as pd

model = pybamm.lithium_ion.DFN(options = {
    "thermal":"lumped",
    "dimensionality":0,
    "cell geometry": "arbitrary", 
    "SEI": "solvent-diffusion limited",
    "SEI porosity change": "true",
    "particle mechanics": ("swelling and cracking", "swelling and cracking"),
    "SEI on cracks": "true",
    "loss of active material": "stress-driven",
    "calculate discharge energy": "true",
    "lithium plating": "partially reversible",
    "lithium plating porosity change": "true"
    },
    name="TDFN",
    )

param = pybamm.ParameterValues("OKane2022")

#Simple cycling experiment
N=20

cccv_experiment = pybamm.Experiment([
    ("Charge at C/2 until 4.2V",
    "Hold at 4.2V until C/20",
    "Discharge at 1C until 2.5V",
    )
    ]*N
    )

charge_experiment = pybamm.Experiment([(
    "Charge at C/2 until 4.2V",
    "Hold at 4.2V until C/20",
    )])

rpt_experiment = pybamm.Experiment([(
    "Discharge at C/3 until 2.5V",
    )])

#Run model

cccv_sols = []
charge_sols = []
rpt_sols = []
M = 11
for i in range(M):
    if i != 0:  # skip the first set of ageing cycles because it's already been done
        sim = pybamm.Simulation(model, experiment=cccv_experiment, parameter_values=param)
        cccv_sol = sim.solve(starting_solution=rpt_sol)
        sim = pybamm.Simulation(model, experiment=charge_experiment, parameter_values=param)
        charge_sol = sim.solve(starting_solution=cccv_sol)
        sim = pybamm.Simulation(model, experiment=rpt_experiment, parameter_values=param)
        rpt_sol = sim.solve(starting_solution=charge_sol)
    cccv_sols.append(cccv_sol)
    charge_sols.append(charge_sol)
    rpt_sols.append(rpt_sol)

#Plot Discharge Capacity

cccv_cycles = []
cccv_capacities = []
rpt_cycles = []
rpt_capacities = []
for i in range (M):
    for j in range(N):
        cccv_cycles.append(i*(N+2)+j+1)
        start_capacity = rpt_sol.cycles[i*(N+2)+j].steps[2]["Discharge capacity [A.h]"].entries[0]
        end_capacity = rpt_sol.cycles[i*(N+2)+j].steps[2]["Discharge capacity [A.h]"].entries[-1]
        cccv_capacities.append(end_capacity-start_capacity)
    rpt_cycles.append((i+1)*(N+2))
    start_capacity = rpt_sol.cycles[(i+1)*(N+2)-1]["Discharge capacity [A.h]"].entries[0]
    end_capacity = rpt_sol.cycles[(i+1)*(N+2)-1]["Discharge capacity [A.h]"].entries[-1]
    rpt_capacities.append(end_capacity-start_capacity)
plt.scatter(cccv_cycles,cccv_capacities,label="Ageing cycles")
plt.scatter(rpt_cycles,rpt_capacities,label="RPT cycles")
plt.xlabel("Cycle number")
plt.ylabel("Discharge capacity [A.h]")
plt.legend()

I keep getting an error of "list index out of range" for the following line of code: "start_capacity = rpt_sol.cycles[i*(N+2)+j].steps[2]["Discharge capacity [A.h]"].entries[0]". In fact, I downloaded the example notebook and tried plugging M=11 and N=20 and I get the exact same error. I don't get the error when I use the notebook values of M=5 and N=10. 

Forgive me for the seemingly simplistic nature of this question, but could anyone tell me why I getting the list index out of range error when I simply try to change the values of M and N? 

My basic purpose here is to plot the discharge capacity over the number of cycles with the rpt portion of my experiment. If there is a simpler way to do this, please do let me know as well. 

Thank you for your time.

**Comment by MirAbbasAli2A** [comment, marked as answer]
Never mind, I figured it out. Was an error with my code.

**Reply by yazyr** [reply]
@MirAbbasAli2A Hi! Could you kindly let me know what was the issue with your code? I'm trying to do a similar experiment and it would be a great favour if you could guide me what the error was? Thanks

**Reply by MirAbbasAli2A** [reply]
Hey, I had to declare rpt_sol and cccv_sol = None before the initial for loop. Then it worked.
